File: SDKs/microsoft.com/cognitiveservices-Training/3.2/java/lib/src/main/java/openapisdk/models/shared/TagFilter.java
package openapisdk.models.shared;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonInclude.Include;
import openapisdk.utils.SpeakeasyMetadata;

/**
 * TagFilter
 * Model that query for counting of images whose suggested tags match given tags and their probability are greater than or equal to the given threshold.
**/
public class TagFilter {
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("tagIds")
@SpeakeasyMetadata("form:name=tagIds")
    public String[] tagIds;
    public TagFilter withTagIds(String[] tagIds) {
        this.tagIds = tagIds;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("threshold")
@SpeakeasyMetadata("form:name=threshold")
    public Double threshold;
    public TagFilter withThreshold(Double threshold) {
        this.threshold = threshold;
        return this;
    }
}

File: SDKs/microsoft.com/cognitiveservices-Training/3.1/python/src/sdk/models/shared/tagfilter.py
import dataclasses
from typing import Optional
from dataclasses_json import dataclass_json
from sdk import utils


@dataclass_json
@dataclasses.dataclass
class TagFilter:
    r"""TagFilter
    Model that query for counting of images whose suggested tags match given tags and their probability are greater than or equal to the given threshold.
    """
    
    tag_ids: Optional[list[str]] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('tagIds') }, 'form': { 'field_name': 'tagIds' }})
    threshold: Optional[float] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('threshold') }, 'form': { 'field_name': 'threshold' }})
    
